skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitGenPresenter.kt

```diff
@@ -0,0 +1,236 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.tooling.projectgen.circuitgen
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateListOf
+import com.slack.circuit.runtime.presenter.Presenter
+import com.squareup.kotlinpoet.ClassName
+import java.nio.file.Path
+import slack.tooling.projectgen.CheckboxElement
+import slack.tooling.projectgen.DividerElement
+import slack.tooling.projectgen.SectionElement
+import slack.tooling.projectgen.TextElement
+import slack.tooling.projectgen.circuitgen.CircuitGenClassNames.Companion.APP_SCOPE
+import slack.tooling.projectgen.circuitgen.CircuitGenClassNames.Companion.USER_SCOPE
+
+internal class CircuitGenPresenter(
+  private val directory: Path,
+  private val packageName: String?,
+  private val baseTestClass: Map<String, String?>,
+  private val fileGenerationListener: FileGenerationListener,
+  private val onDismissDialog: () -> Unit,
+  private val generationMode: GenerationMode,
+) : Presenter<CircuitGenScreen.State> {
+
+  private val featureName = TextElement("", "Feature Name")
+  private val circuitUi =
+    CheckboxElement(false, name = "UI", hint = "Generate UI class", indentLevel = 10)
+  private val circuitPresenter =
+    CheckboxElement(false, name = "Presenter", hint = "Generate Presenter class", indentLevel = 10)
+  private val assistedInjection =
+    SectionElement("Assisted Injection", "(Optional)", indentLevel = 10)
+  private val assistedScreen =
+    CheckboxElement(
+      false,
+      name = "Screen",
+      hint = "Add @assisted for screen parameter",
+      indentLevel = 30,
+    )
+  private val assistedNavigator =
+    CheckboxElement(
+      false,
+      name = "Navigator",
+      hint = "Add @assisted for navigator parameter",
+      indentLevel = 30,
+    )
+  private val circuitInject = SectionElement("Circuit Injection", "(Optional)", indentLevel = 10)
+  private val userScopeCircuitInject =
+    CheckboxElement(
+      false,
+      name = "User Scope",
+      hint = "Add UserScope to Circuit Inject",
+      indentLevel = 30,
+    )
+  private val appScopeCircuitInject =
+    CheckboxElement(
+      false,
+      name = "App Scope",
+      hint = "Add AppScope to Circuit Inject",
+      indentLevel = 30,
+    )
+  private val test =
+    CheckboxElement(false, name = "Tests", hint = "Should generate test class", indentLevel = 10)
+
+  private val screenAndPresenterElements =
+    mutableStateListOf(
+      featureName,
+      DividerElement,
+      SectionElement("Class(es) to generate", ""),
+      circuitUi,
+      circuitPresenter,
+      assistedInjection,
+      assistedScreen,
+      assistedNavigator,
+      circuitInject,
+      userScopeCircuitInject,
+      appScopeCircuitInject,
+      test,
+    )
+  private val viewModelElements =
```

Comment:
I don't think we should support viewmodel generation here. It's not useful outside of slack's codebase and also not something we're recommending people write new ones of anyway

Reply:
removed ViewModel generation from this PR https://github.com/slackhq/slack-gradle-plugin/pull/799/commits/25eaeafd2f23ad70c112008154821f4ab7c6420b